### PR TITLE
Script cleanups

### PR DESCRIPTION
**Pull request by tomchristie**
* Dropping `PYTHONPATH`, since we've got a `-e .` in the requirements.txt anyway.
* Adding `black` and `flake8` to the checks.
* Consistent `PREFIX` style everywhere.
* Dropping `setup.cfg` in favour of arguments in the scripts.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-include_trailing_comma = True
-known_first_party = httpcore,tests
-known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytest,rfc3986,setuptools,sniffio,trio,trustme,urllib3,uvicorn
-line_length = 88
-multi_line_output = 3
-
-[tool:pytest]
-addopts = --cov=httpcore --cov=tests --cov-report=term-missing -rxXs
-markers =
-  copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup
```
Comment (yeraydiazdiaz) [inline setup.cfg:21]: This file is actually quite useful for me as it's read by the editor when linting and configures the tools when I invoke them myself (usually just `pytest` but still). I also find it nicer to have all the tools config in a single, more readable place as some of them can be quite long.

I'd prefer keeping it, if that's ok?

File: scripts/lint [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+    PREFIX="venv/bin/"
+else
+    PREFIX=""
 fi
-export SOURCE_FILES="httpcore tests"
 
 set -x
 
-${PREFIX}autoflake --in-place --recursive $SOURCE_FILES
-${PREFIX}isort --project=httpcore --recursive --apply $SOURCE_FILES
```
Comment (yeraydiazdiaz) [inline scripts/lint:12]: `isort` is removed from here but not added to `check`, should it be there or are we dropping it?

Comment (tomchristie) [reply scripts/lint:12]: I'm not sure actually. There's a little niggle with it at the moment, in that `isort` seems to be making one adjustment that black then reverts, which means we get noisy outputs even when everything is clean.

File: setup.cfg (hunk 1 above)
Comment (tomchristie) [reply setup.cfg:21]: Possibly.

I'm not wild about `setup.cfg` myself, and keen on removing top-level config files from the repo where possible, but it's certainly worth a conversation about what we're happy to have across all the encode repos as a whole.

I'm defo happy to drop this from the PR, and keep any changes there scoped to another conversion.

File: scripts/lint (hunk 2 above)
Comment (yeraydiazdiaz) [reply scripts/lint:12]: I've noticed that too but seems we end up with a pretty consistent setup, albeit a noisy one.

I'm not too fussed either way, but if we do decide to drop `isort` we probably want to remove it from the requirements as well.

File: setup.cfg (hunk 1 above)
Comment (florimondmanca) [reply setup.cfg:21]: Agreed with Yeray on this - these config files are super useful as IDEs that automatically read them without having to change anything in project settings.

File: scripts/lint [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-${PREFIX}black --exclude 'httpcore/_sync/.*' --exclude 'tests/sync_tests/.*' --target-version=py36 $SOURCE_FILES
+${PREFIX}black --exclude '(httpcore/_sync|tests/sync_tests)/.*' --target-version=py37 ${PACKAGE} tests
```
Comment (florimondmanca) [inline scripts/lint:13]: Thoughts on keeping `autoflake`? If we're adding `flake8` to `check` it's definitely nice to have a quick way to fix things when they're broken.

**Comment by yeraydiazdiaz** [comment]
@euri10 wrong button?

**Comment by euri10** [comment]
Phone in my pocket 🙄 sorry

**Pull request closed without merging**